src/inventory.h

```diff
@@ -102,18 +102,25 @@ struct ItemStack
 	}
 
 	// Get tool digging properties, or those of the hand if not a tool
+	// If not hand assumes default hand ""
 	const ToolCapabilities& getToolCapabilities(
-			const IItemDefManager *itemdef) const
+			const IItemDefManager *itemdef, const ItemStack* hand = nullptr) const
 	{
-		const ToolCapabilities *item_cap =
-			itemdef->get(name).tool_capabilities;
-
-		if (item_cap == NULL)
-			// Fall back to the hand's tool capabilities
-			item_cap = itemdef->get("").tool_capabilities;
-
-		assert(item_cap != NULL);
-		return metadata.getToolCapabilities(*item_cap); // Check for override
+		const ToolCapabilities *item_cap = itemdef->get(name).tool_capabilities;
+
+		if (item_cap != nullptr) {
```

Comment:
```suggestion
		if (item_cap) {
```

same below, just check the pointer directly for truthiness (same below)